aws/resource_aws_vpn_connection.go

```diff
@@ -277,6 +516,182 @@ func resourceAwsVpnConnectionCreate(d *schema.ResourceData, meta interface{}) er
 		{}, {},
 	}
 
+	if v, ok := d.GetOk("tunnel1_dpd_timeout_action"); ok {
```

Comment:
Since there are so many options here, it would make it easier to read to break all of these out into an expand function (and maybe more). See https://github.com/hashicorp/terraform-provider-aws/blob/master/aws/resource_aws_networkfirewall_rule_group.go#L573 for an example.

Reply:
Done